modal/app.py

```diff
@@ -478,11 +478,14 @@ def _get_default_image(self):
             return _default_image
 
     def _get_watch_mounts(self):
+        if not self._running_app:
+            raise InvalidError("`_get_watch_mounts` requires a running app.")
```

Comment:
Should we be referencing a private method in a user-facing error? Can we catch this earlier?

Reply:
This is not supposed to be a user facing error, so maybe InvalidError is actually a bit misleading. Should basically never happen unless we mess up in the code calling `_get_watch_mounts` (which is only used by `modal serve`)